src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10411,30 +10415,79 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
                         if (m.IsStatic) continue;
                         break;
                 }
+                methods.Add(m);
+            }
+
+            if (!OverloadResolution.FilterMethodsForUniqueSignature(methods, out useParams))
+            {
+                methods.Free();
+                return null;
+            }
+
+            var seenAnyApplicableCandidates = methods.Count != 0;
+
+            foreach (var m in methods)
+            {
                 if (!isCandidateUnique(ref method, m))
                 {
+                    methods.Free();
                     return null;
                 }
             }
+
             if (node.SearchExtensionMethods)
             {
                 var receiver = node.ReceiverOpt!;
                 foreach (var scope in new ExtensionMethodScopes(this))
                 {
+                    methods.Clear();
                     var methodGroup = MethodGroup.GetInstance();
                     PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, node.Syntax, receiver, node.Name, node.TypeArgumentsOpt, BindingDiagnosticBag.Discarded);
                     foreach (var m in methodGroup.Methods)
                     {
-                        if (m.ReduceExtensionMethod(receiver.Type, Compilation) is { } reduced &&
-                            !isCandidateUnique(ref method, reduced))
+                        if (m.ReduceExtensionMethod(receiver.Type, Compilation) is { } reduced)
                         {
-                            methodGroup.Free();
-                            return null;
+                            methods.Add(reduced);
```

Comment:
>methods.Add(reduced); [](http://example.com/codeflow?start=28&length=21)

It looks like we might get in trouble with reduced form in ```FilterMethodsForUniqueSignature``` with respect to determining whether the extension method can be consumed in expanded form. ```IsValidParams``` relies on type from original parameter's definition. Parameters of reduced extension methods say that they are definitions, but reduction might have changed their types. Therefore, the parameter might appear as a valid `params`, when the parameter in the original extension method isn't, even when it has `params` modifier (can come from metadata, an erroneous declaration in source possibly can be in this state too).  Compiler never consumes reduced extension methods, they are never given to overload resolution when we bind an invocation. Compiler always binds against unreduced forms of extension methods.
Let's try creating a unit test confirming the problem.

Reply:
> Let's try creating a unit test confirming the problem.

I wasn't able to create such unit test. `IsValidParams` relies on `TryInferParamsCollectionIterationType` to which it passes the original definition (`final.OriginalDefinition.Type`), i.e., even if one creates an overload with `params T arg` which after reducing becomes `params int[] arg` or `params List<int> arg`, the `TryInferParamsCollectionIterationType` only sees the original `T` and returns `false`.

Reply:
> I wasn't able to create such unit test.

Is there a test that demonstrates what you described?

Reply:
Added `Params_ReducedExtensionsMethod_*` tests.

Reply:
>  [reduced extension methods] are never given to overload resolution when we bind an invocation

Would it be valuable to assert that overload resolution is never given reduced extension methods?